java/sage/Carny.java

```diff
@@ -1159,61 +1617,87 @@ private void stdProcessing()
     newWPMap.keySet().removeAll(watchedPotsToClear);
 
     if (Sage.DBG) System.out.println("CARNY Negative Energy Size: " + (blackBalled.size() + stoners.length));
-    for (Airing badAir : blackBalled)
+    for (int i = 0, blackBalledSize = blackBalled.size(); i < blackBalledSize; i++)
     {
+      Airing badAir = blackBalled.get(i);
```

Comment:
Shouldn't the last line of this loop now be 'firstCallback.removeAirSet(badAir)' for efficiency?